src/lib.rs

```diff
@@ -68,6 +76,25 @@ async fn nixci_build(
             );
         }
     }
+
+    if build_cfg.print_all_dependencies {
+        let mut all_drvs = Vec::new();
+        for out in all_outs.iter() {
+            if let DrvPaths::DrvOut(drv_out) = out {
+                let drv = nix::all_deps::nix_store_query_deriver(drv_out.clone()).await?;
+                all_drvs.push(drv);
+            }
+        }
+        for drv in all_drvs {
+            let deps = nix::all_deps::nix_store_query_requisites_with_outputs(drv.clone()).await?;
+            all_outs.extend(deps.into_iter());
+        }
```

Comment:
FWIW, it is not immediately clear what this does (ie., without reading the implementation). But in any case, this whole code should probably be made a (pure) function with clear name/ docs, and be moved to `nix_store.rs`.